### PR TITLE
bump akka version for 2.11+, introduce AkkaBackCompat

### PR DESCRIPTION
**Pull request by sugakandrey**
Akka keeps writing to `STDOUT` on shutdown in 2.4.12. We may want to wait until we find a way to fix this before merging.

### CODE REVIEW

**Comment by fommil** [comment]
@bwmcadams said he may look into this

File: core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala [hunk 1]
```diff
@@ -17,8 +18,8 @@ trait IsolatedSearchServiceFixture extends IsolatedSourceResolverFixture {
       testCode(config, searchService)
     } finally {
       Await.ready(searchService.shutdown(), Duration.Inf)
-      actorSystem.shutdown()
-      actorSystem.awaitTermination(10.seconds)
+      actorSystem.close()
```
Comment (fommil) [inline core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: I was thinking you could make the BackCompat only for 2.10 and use the akka 2.4 names... that way you don't need to write an impl for 2.11+

File: project/SensibleSettings.scala [hunk 2]
```diff
@@ -143,15 +151,15 @@ object Sensible {
     "com.google.code.findbugs" % "jsr305" % "3.0.1" % "provided"
   )
 
-  def testLibs(config: String = "test") = Seq(
+  def testLibs(config: String = "test") = Def.setting(Seq(
```
Comment (fommil) [inline project/SensibleSettings.scala]: I will probably remove a bunch of stuff from this `testLibs` when I publish it as a separate plugin

File: core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala (hunk 1 above)
Comment (sugakandrey) [inline core/src/it/scala/org/ensime/fixture/SearchServiceFixture.scala]: Makes sense. Will do.

**Comment by sugakandrey** [comment]
damn, forgot we have `-Ywarn-unused-import` & `-Xfatal-warnings`

**Comment by sugakandrey** [comment]
@fommil what should I do about this? Having `BackCompat` only for 2.10 forces me to have an unused import in 2.11 code.

**Comment by fommil** [comment]
you could make it a trait instead of an object, and mix it in. Sorry while I go and barf somewhere...

**Comment by fommil** [comment]
We have defined timeouts in EnsimeSpec for akka that get scaled by envvars, might be best to use those.

**Comment by fommil** [comment]
Do we really care about waiting for shutdown in tests?

**Comment by fommil** [comment]
I guess we do, otherwise it might never work.

**Comment by fommil** [comment]
On those stdouts, I'm tempted to just monkey patch whatever part of akka is doing it.

**Comment by sugakandrey** [comment]
@fommil funny thing, our shutdown tests seem to have been broken all along
```
eventually(timeout(30 seconds), interval(1 second)) {
          system.isTerminated
}
```
This block never fails, since `eventually` succeeds whenever block passed to it returns a value, doesn't matter if it is `true` or `false` or whatever else.

**Comment by fommil** [comment]
very interesting, this might be what @dragos was seeing

**Comment by sugakandrey** [comment]
Currently our `WebServer` binds to `/0.0.0.0:port` and not `/127.0.0.1:port`. See https://github.com/ensime/ensime-server/blob/2.0/server/src/main/scala/org/ensime/server/WebServer.scala#L55. Is this intentional?

**Comment by fommil** [comment]
nope, should definitely be 127.0.0.1, well spotted!

**Comment by sugakandrey** [comment]
Alright, so this should fix `WebServer` startup address. Also we failed to report bind failures in [WebServer](https://github.com/ensime/ensime-server/blob/2.0/server/src/main/scala/org/ensime/server/WebServer.scala#L56-L64), which resulted in [this code](https://github.com/ensime/ensime-server/blob/2.0/server/src/main/scala/org/ensime/server/Server.scala#L75-L90) going into `Success` branch even if `Server` failed to bind and then dying with `NullPointerException` since `ch.localAddress()` was `null`. Fixed that too.

**Comment by fommil** [comment]
cool! I just merged a huge PR today, can you please rebase?

**Comment by dragos** [comment]
All right, glad to see I'm not the only one seeing zombies.

**Comment by sugakandrey** [comment]
Sure, I'll rebase, anyways we need a way to deal with akka logging.

**Pull request merged**